Bump version of ffi-rzmq to latest

mgatny opened the pull request:
- [x] Make dealer sock work with receive (not just publish)

mgatny committed:
Bump version of ffi-rzmq to latest

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/connamara/agent_zmq?pullRequest=11) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/connamara/agent_zmq?pullRequest=11) before we can accept your contribution.<br/>

mgatny commented:
@cbusbey I move that we drop ruby 1.9.2 from travis and add one of more modern versions

mgatny committed:
Fixup old typo in gemspec

mgatny committed:
Ran `rake gemspec` to update gemspec

mgatny committed:
Make dealer agent a MessageCache so it can receive

mgatny committed:
Make shift an alias of pop so dealer can recv

mgatny committed:
Make rspec expect be_truthy instead of be_true

mgatny committed:
Set sock opts before/after bind/connect as needed

Based on the zmq_setsockopt docs:
Caution: All options, with the exception of ZMQ_SUBSCRIBE, ZMQ_UNSUBSCRIBE and ZMQ_LINGER,
only take effect for subsequent socket bind/connects.

mgatny committed:
Set no linger on sock close. Kill context on stop.

mgatny committed:
Remove old ruby from travis cfg, add new ruby

mgatny committed:
Update ruby versions
